db/db_impl/db_impl_open.cc

```diff
@@ -973,6 +980,9 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& log_numbers,
         status = Status::OK();
         stop_replay_for_corruption = true;
         corrupted_log_number = log_number;
+        if (corrupted_log_found != nullptr) {
```

Comment:
Should this be here or at line# 837? IIUC, we need to insert a dummy record only if the newest log is corrupted, right?

Reply:
Line #837 is reached when we have a double crash, and the logs after corrupted one are created after the 1st recovery. This is actually a case that we don't need to insert dummy since the continuity of sequence numbers are already taken care of in the 1st recovery.
You are right that the cases that do insert the dummy record could be more limited but limiting it would not bring any performance advantage yet bring more complexity on reasoning which case is which. It is simpler to always start the first log after recovery with a dummy record if there is a corrupted log that we have skipped.